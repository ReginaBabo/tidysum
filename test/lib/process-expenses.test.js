const expect = require('expect');
const toMatchSnapshot = require('expect-mocha-snapshot');
const processExpenses = require('../../lib/process-expenses');

expect.extend({ toMatchSnapshot });

describe('processExpenses', () => {
  describe('processFile', () => {
    it('Processes a small file with two months in each of two years', async function() {
      let result = await processExpenses.processFile(
        `${process.cwd()}/test/fixtures/small-test-data.csv`
      );
      expect(JSON.stringify(result, null, 2)).toMatchSnapshot(this);
    });

    it('Processes demo data', async function() {
      let result = await processExpenses.processFile(
        `${process.cwd()}/test/fixtures/demo-data.csv`
      );
      expect(JSON.stringify(result, null, 2)).toMatchSnapshot(this);
    });
  });
});
