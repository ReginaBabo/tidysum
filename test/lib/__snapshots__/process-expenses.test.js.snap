// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processExpenses processFile Processes a small file with two months in each of two years 1`] = `
"{
  \\"2017\\": {
    \\"total\\": \\"50\\",
    \\"Jan\\": {
      \\"total\\": \\"30\\",
      \\"byCategory\\": {
        \\"Clothing\\": \\"10\\",
        \\"Food\\": \\"20\\"
      },
      \\"byMerchant\\": {
        \\"Merchant A\\": \\"10\\",
        \\"Merchant B\\": \\"20\\"
      }
    },
    \\"Mar\\": {
      \\"total\\": \\"20\\",
      \\"byCategory\\": {
        \\"Clothing\\": \\"10\\",
        \\"Entertainment\\": \\"10\\"
      },
      \\"byMerchant\\": {
        \\"Merchant A\\": \\"10\\",
        \\"Merchant C\\": \\"10\\"
      }
    }
  },
  \\"2018\\": {
    \\"total\\": \\"70\\",
    \\"Jan\\": {
      \\"total\\": \\"40\\",
      \\"byCategory\\": {
        \\"Clothing\\": \\"15\\",
        \\"Food\\": \\"25\\"
      },
      \\"byMerchant\\": {
        \\"Merchant A\\": \\"15\\",
        \\"Merchant B\\": \\"25\\"
      }
    },
    \\"Mar\\": {
      \\"total\\": \\"30\\",
      \\"byCategory\\": {
        \\"Clothing\\": \\"15\\",
        \\"Entertainment\\": \\"15\\"
      },
      \\"byMerchant\\": {
        \\"Merchant A\\": \\"15\\",
        \\"Merchant C\\": \\"15\\"
      }
    }
  }
}"
`;

exports[`processExpenses processFile Processes demo data 1`] = `
"{
  \\"2018\\": {
    \\"total\\": \\"239.94\\",
    \\"Oct\\": {
      \\"total\\": \\"228.27\\",
      \\"byCategory\\": {
        \\"Groceries\\": \\"88.29\\",
        \\"Restaurant\\": \\"133.99\\",
        \\"Entertainment\\": \\"5.99\\"
      },
      \\"byMerchant\\": {
        \\"Loblaws\\": \\"88.29\\",
        \\"The Keg\\": \\"133.99\\",
        \\"Amazon\\": \\"5.99\\"
      }
    },
    \\"Nov\\": {
      \\"total\\": \\"11.67\\",
      \\"byCategory\\": {
        \\"Health\\": \\"11.67\\"
      },
      \\"byMerchant\\": {
        \\"Loblaws\\": \\"11.67\\"
      }
    }
  }
}"
`;
